### PR TITLE
ShuffleStrings uses a seeded rand object

### PR DESCRIPTION
**Pull request by ArtfulCoder**

### CODE REVIEW

**Comment by ArtfulCoder** [comment]
Fixes the issue raised in #13619

File: pkg/util/rand/rand.go [hunk 1]
```diff
@@ -54,3 +54,11 @@ func Seed(seed int64) {
 
 	rng.rand = rand.New(rand.NewSource(seed))
 }
+
+// Perm returns, as a slice of n ints, a pseudo-random permutation of the integers [0,n)
+// from the default Source.
+func Perm(n int) []int {
+	rng.Lock()
```
Comment (thockin) [inline pkg/util/rand/rand.go]: Who seeds this RNG?  I don't see a call to Seed() in here - isn't this the same problem, just one extra layer?

Comment (ArtfulCoder) [inline pkg/util/rand/rand.go]: wouldn't this be enough ? https://github.com/ArtfulCoder/kubernetes/blob/shufflestrings/pkg/util/rand/rand.go#L32

Comment (thockin) [inline pkg/util/rand/rand.go]: oops.  I missed that.  Was too busy looking for init()

**Comment by thockin** [comment]
LGTM

**Comment by ArtfulCoder** [comment]
e2e is flaky. it was green in the previous run.
should be safe to merge

**Comment by thockin** [comment]
@k8s-bot test this please

**Pull request merged**